x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -958,7 +958,11 @@ export function overridePackageInputs(
     }
 
     if (override.vars) {
-      originalInput = deepMergeVars(originalInput, override);
+      if (!originalInput.vars) {
+        originalInput.vars = { ...override.vars };
+      } else {
+        originalInput = deepMergeVars(originalInput, override) as NewPackagePolicyInput;
```

Comment:
I think the function deepMergeVars should probably set the `originalnput.vars` you will have to fix it at only one place and it's probably more future proof if we have vars at another place in the future. What do you think>

Reply:
Good thought. I will make this change.

Reply:
Addressed in https://github.com/elastic/kibana/pull/110698/commits/7ca7744bb25a36cb55057a680a3474f82e217aa3